src/orquestra/opt/optimizers/simple_gradient_descent.py

```diff
@@ -62,22 +67,55 @@ def _minimize(
                 see note.
 
         """
-        assert isinstance(cost_function, CallableWithGradient)
-
+        # So that mypy does not complain about missing attributes:
+        assert hasattr(cost_function, "gradient")
```

Comment:
Hmmm mypy shouldn't complain here. Might be worth talking to @dexter2206 about this.

Reply:
So maybe it's just my linter? I think I'm using pylance or something like that.